### DISCUSSION TITLE
Does Tabular Editor moves any Data to external Servers while executing DAX queries

### DISCUSSION THREAD

**Discussion by moeeljawadoph** [TabularEditor/TabularEditor3#919] [Other Topics]
Hello, 
From what i have understood of several YouTube videos introducing TabularEditor:
- the Data in the Editor still resides on a Power Bi instance,
- this Power BI instance is used during queries execution.

I would like to confirm that when i am connecting Power BI to TabularEditor, there is **no data being stored/transferred** outside of Power BI (or on TabularEditor side).

Thanks in Advance

**Comment by otykier** [comment]
Hi,
The data you see in Tabular Editor's DAX query results view has been transferred from Power BI to Tabular Editor (otherwise, how would Tabular Editor be able to show the data?)

However, Tabular Editor does not persist this data anywhere - as soon as you close the query window in Tabular Editor, the data is gone. Users of Tabular Editor may still copy/paste the data using the Windows clipboard, or take screenshots, etc., but this is no different from when you access a Power BI dataset using other client tools.

**Reply by moeeljawadoph** [reply]
Thanks for your reply, Really appreciated.
My question is more about the security aspect of Tabular Editor. From your second paragraph, i can understand that the data being accessed from my machine, is not transferred anywhere outside of my machine or (powerbi.com), the read of data, execution of dax queries is not happening any where on a remote cloud service provided by Tabular Editor.
Thanks in advance

**Reply by mlonsk** [reply]
Hi @moeeljawadoph 
To close off the topic here, your statement is correct. Tabular Editor does not transfer or persist the data outside of the connection between your machine and the PowerBI service. 

And apologies for the late reply.
